### DISCUSSION TITLE
Give Regex pins for set_expression and set_output

### DISCUSSION THREAD

**Discussion by Tsunder** [FakeFishGames/Barotrauma#5172] [Ideas and suggestions]
signal checks have set_target and set_output, regex component should have equivalent too!
set_expression to set the expression of the regex component
set_output to set the output of the regex component

**Comment by dhdoctor** [comment]
Devs please god I beg of you I need this functionality. I've been designing an OS for my sub for months and this would MASSIVELY help me. I love this game to absolute bits! I love wiring for hours on end! This would just further my addiction. I plead on my knees for you to head my prayers.

**Reply by SomeRandomNoobKekeke** [reply]
Why don't you use lua or c# components? Regex is the worst thing to build OS on

You also would need to improve regex protection against too slow queries https://github.com/FakeFishGames/Barotrauma/discussions/14192

**Reply by dhdoctor** [reply]
Building it in a campaign with friends had no issues as of yet and it's about 8 boxes big so far. Everyone I play with is very adamant about playing vanilla plus I don't know lua or c# and have just started understanding regex. Would apply to many other wire projects I do though not just my os. I was astonished to see someone has addressed it with the exact same arguments I would put forth for it. It seems like a logical pin in to include and would open up functionality in vanilla a little more for the engineer redstone freak types (which i feel is the embodiment of engineer in this game).
